hs_file_types/models/base.py

```diff
@@ -998,33 +954,20 @@ def supports_unzip(self):
     @property
     def aggregation_name(self):
         """Returns aggregation name as per the aggregation naming rule defined in issue#2568"""
-        if self.is_single_file_aggregation:
-            # self is a single file aggregation type
-            return self.files.first().short_path
-        else:
-            # self is a multi- file aggregation type
-            if not self.is_fileset:
-                return self.files.first().file_folder
+        if not self.is_fileset:
+            # any aggregation that is not a fileset type, the path of the aggregation primary file
+            # is the aggregation name
+            primary_file = self.get_primary_resouce_file(self.files.all())
+            return primary_file.short_path
 
-            # self is a fileset aggregation
-            return self.folder
+        # self is a fileset aggregation - aggregation folder path is the aggregation name
+        return self.folder
```

Comment:
The concept of a folder for aggregation name is gone now and replaced with dataset_name.  It'd be nice to not make an exception for fileset either, we should have the same rules apply across all aggregations.  I'll spend some time working on this today and see if I can get it all to work together.

Reply:
Using dataset_name for aggregation name would work only if we enforce dataset_name to be unique within a resource. Also the design specification for aggregation defined aggregation name to be a file/folder path  which would allow aggregation to be identified using aggregation name instead of id of the aggregation. I guess we can use dataset_name instead of aggregation id if we make dataset_name unique within a resource. May be I am missing something..

Reply:
the logic I had in there included the path to the aggregation, so it does make it unique.  folder no longer makes sense because there can be multiple aggregations inside a folder now.

Reply:
Given that there is no longer a physical folder, the aggregation name needs to be thought of as a URI/URL that identifies the aggregation. As @sblack-usu suggests, including the full URL path to the aggregation that ends with the aggregation name would work right?

Reply:
@horsburgh I assume we still have folder for fileset.

Reply:
As long as we are using a physical path for aggregation name it should work - that's how it is currently running on www. Use of dataset_name for aggregation name won't work.

Reply:
For a fileset aggregation the aggregation name is the folder path. For any other aggregation types, the path of the primary file of that aggregation is the aggregation name.

Reply:
yes, I left folder in for fileset

Reply:
We don't have a physical path for aggregation anymore, the primary file won't work anymore because it's valid to request the primary file alone (I'm thinking of the download code we need to be able to tell the difference between a request for the file and a request for the aggregation). 

It makes sense to me to use the path + dataset_name to identify an aggregation request (that is consistent with how we portray an aggregation to the user in the ui).  A fileset will be left alone and identified by the folder.